subprojects/base-services/src/main/java/org/gradle/internal/work/DefaultConditionalExecutionQueue.java

```diff
@@ -78,10 +78,16 @@ public void submit(ConditionalExecution<T> execution) {
 
     @Override
     public void expand() {
+        expand(false);
+    }
+
+    private void expand(boolean force) {
```

Comment:
⭕️ This is kinda hard to understand and could use some (class/method-level) documentation

Reply:
Yeah, good point.  I've added some comments to (hopefully) make the function of this thing a little clearer.